infer_segmentation.py

```diff
@@ -0,0 +1,346 @@
+                prediction_threshold[i] = threshold_output_data['metadata']['thresholds_training_output'][str('class'+str(i))]['x']
+            else:
+                AssertionError('Unsuccessfully trained threshold attempted to be loaded.')
+    else:
+        prediction_threshold = np.ones(num_classes) * global_threshold
+
+    compiled_model = generate_compiled_segmentation_model(
+        train_config['segmentation_model']['model_name'],
+        train_config['segmentation_model']['model_parameters'],
+        num_classes,
+        train_config['loss'],
+        train_config['optimizer'],
+        Path(local_model_dir, "model.hdf5").as_posix(),
+        optimized_class_thresholds=optimized_class_thresholds)
+
+    if image_ids is None:
+        images_list = list(Path(image_folder).iterdir())
+    else:
+        images_list = image_ids.split(',')
+
+    labels_output = str2bool(labels_output)
+
+    pad_output = str2bool(pad_output)
+
+    n_images = len(list(Path(image_folder).iterdir()))
+    for i, image_file in enumerate(sorted(Path(image_folder).iterdir())):
+        if image_file.parts[-1] in images_list:
+            print('Segmenting image {} of {}...'.format(i, n_images))
+
+            image = Image.open(image_file)
+
+            segmented_image = segment_image(compiled_model, image, prediction_threshold,
+                                            target_size_1d, background_class_index, labels_output, pad_output)
+
+            # enable saving of various versions of same inference
+            image_file_ext = image_file.parts[-1].split('.')[-1]
+            if labels_output and pad_output:
+                segmented_image.save(Path(output_dir, str(
+                    image_file.parts[-1].split('.')[0] + '_pad_labels' + '.'
+                    + image_file_ext)).as_posix())
+            elif labels_output:
+                segmented_image.save(Path(output_dir, str(
+                    image_file.parts[-1].split('.')[0] + '_labels' + '.'
+                    + image_file_ext)).as_posix())
+            elif pad_output:
+                segmented_image.save(Path(output_dir, str(
+                    image_file.parts[-1].split('.')[0] + '_pad' + '.'
+                    + image_file_ext)).as_posix())
+            else:
+                segmented_image.save(Path(output_dir, str(
+                    image_file.parts[-1].split('.')[0] + '.'
+                    + image_file_ext)).as_posix())
+
+    metadata = {
+        'gcp_bucket': gcp_bucket,
+        'model_id': model_id,
+        'user_specified_prediction_thresholds': user_specified_prediction_thresholds,
+        'trained_thresholds_id': trained_thresholds_id,
+        'trained_class_thresholds_loaded': optimized_class_thresholds,
+        'default_global_threshold_for_reference': global_threshold,
+        'prediction_thresholds_used': list(prediction_threshold),
```

Comment:
@Josh-Joseph this is throwing a yaml `safe_dump` representer error. I need to do `str(list())` to get this to pass. It didn't work without str